src/Data/BodyData.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Silktide\ProspectClient\Data;
+
+use Silktide\ProspectClient\Api\Fields\AbstractApiFields;
+
+use function \GuzzleHttp\json_encode;
+
+class BodyData extends AbstractData
+{
+    public function __toString(): string
+    {
+        if($this->count() === 0) {
```

Comment:
As a general rule, I find it's better not to use `__toString()` under the [Ocramius view of thinking on the topic](https://github.com/ShittySoft/symfony-live-berlin-2018-doctrine-tutorial/pull/3#issuecomment-460441229)

Additionally, I've found after doing far too much refactoring of legacy code, it's generally a pain to deal with long term. As typecasts can happen all around the codebase, (dammit PHP :P) it becomes very difficult to definitely prove that a `__toString` function is no longer in use, whereas with a `->toString()` method you get the friendly "hey, is this function being used?" functionality in PHPStorm.